AnaProd/anaTupleProducer.py

```diff
@@ -99,6 +101,8 @@ def addAllVariables(dfw, syst_name, isData, trigger_class):
                                                         -1.f;""")
             dfw.DefineAndAppend(f"tau{leg_idx+1}_gen_vis_mass", f"""tau{leg_idx+1}_genMatchIdx>=0? static_cast<float>(genLeptons.at(tau{leg_idx+1}_genMatchIdx).visibleP4().M()) :
                                                     -1.f;""")
+            dfw.DefineAndAppend(f"tau{leg_idx+1}_gen_vis_e", f"""tau{leg_idx+1}_genMatchIdx>=0? static_cast<float>(genLeptons.at(tau{leg_idx+1}_genMatchIdx).visibleP4().E()) :
+                                                    -1.f;""")
```

Comment:
```suggestion
```

Reply:
for the moment, until the mistery of b-jet discrepancies is totally fixed, I prefer to store at least bjet energy and at this point also taus one.